contributors/devel/pull-requests.md

```diff
@@ -1,111 +1,326 @@
 
+<!-- BEGIN MUNGE: GENERATED_TOC -->
+- [Pull Request Process](#pull-request-process)
+- [Before You Submit a PR](#before-you-submit-a-pr)
+  * [Run Local Verifications](#run-local-verifications)
+  * [Sign the CLA](#sign-the-cla)
+- [The PR Submit Process](#the-pr-submit-process)
+  * [Write Release Notes if Needed](#write-release-notes-if-needed)
+  * [The Testing and Merge Workflow](#the-testing-and-merge-workflow)
+  * [Comment Commands Reference](#comment-commands-reference)
+  * [Automation](#automation)
+  * [How the e2e Tests Work](#how-the-e2e-tests-work)
+- [Why was my PR closed?](#why-was-my-pr-closed-)
+- [Why is my PR not getting reviewed?](#why-is-my-pr-not-getting-reviewed-)
+- [Best Practices for Faster Reviews](#best-practices-for-faster-reviews)
+  * [0. Familiarize yourself with project conventions](#0-familiarize-yourself-with-project-conventions)
+  * [1. Is the feature wanted? Make a Design Doc or Sketch PR](#1-is-the-feature-wanted--make-a-design-doc-or-sketch-pr)
+  * [2. Smaller Is Better: Small Commits, Small PRs](#2-smaller-is-better--small-commits--small-prs)
+  * [3. Open a Different PR for Fixes and Generic Features](#3-open-a-different-pr-for-fixes-and-generic-features)
+  * [4. Comments Matter](#4-comments-matter)
+  * [5. Test](#5-test)
+  * [6. Squashing and Commit Titles](#6-squashing-and-commit-titles)
+  * [7. KISS, YAGNI, MVP, etc.](#7-kiss--yagni--mvp--etc)
+  * [8. It's OK to Push Back](#8-it-s-ok-to-push-back)
+  * [Common Sense and Courtesy](#common-sense-and-courtesy)
 <!-- END MUNGE: GENERATED_TOC -->
 
-# Pull Request Process
+# Before You Submit a PR
+
+This guide is for contributors who already have a PR to submit. If you're looking for information on setting up your developer environment and creating code to contribute to Kubernetes, see the [development guide](development.md). 
+
+**Make sure your PR adheres to our best practices. These include following project conventions, making small PRs, and commenting thoroughly. Please read the more detailed section on [Best Practices for Faster Reviews](#best-practices-for-faster-reviews) at the end of this doc.**
+
+## Run Local Verifications
+
+Run these local verifications before you submit your PR.
+
+* Enable, run, and pass the Kubernetes [pre-commit hook](development.md#define-a-pre-commit-hook) which checks your repo formatting (and more); note that you may want to add these when you're closer to the end of your project, as they take time to run every time you commit
+* Run and pass `make verify` (can take 30-40 minutes)
+* Run and pass `make test`
+* Run and pass `make test-integration`
+
+## Sign the CLA
+
+You must sign the CLA before your first contribution. [Read more about the CLA.](https://github.com/kubernetes/community/blob/master/CLA.md)
+
+If you haven't signed the Content License Agreement (CLA) before making a PR, the `k8s-ci-robot` will leave a comment with instructions on how to sign the CLA.
```

Comment:
This is actually the "Contributor License Agreement" (ref: https://github.com/kubernetes/community/blob/master/CLA.md) not a Content License Agreement